pkg/common/util/dial.go

```diff
@@ -0,0 +1,79 @@
+package util
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+// BlockingDial is a helper method to dial the given address, using optional TLS credentials,
+// and blocking until the returned connection is ready. If the given credentials are nil, the
+// connection will be insecure (plain-text).
+// (This function was borrowed from https://github.com/fullstorydev/grpcurl/blob/master/grpcurl.go)
+func BlockingDial(ctx context.Context, network, address string, creds credentials.TransportCredentials, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
```

Comment:
This function may be better named something like `GrpcBlockingDial` to indicate that it is gRPC-specific